src/components/Cards/FilteredCards.tsx

```diff
@@ -0,0 +1,49 @@
+import React, { FC } from "react";
+import { useAppSelector } from "@hooks/hooks";
+import Card from "@components/Card/Card";
+import { ICard } from "@models/ICard";
+
+const FilteredCards: FC = () => {
+  const searchedValue = useAppSelector((state) => state.food.searchValue);
+  const filterValues = useAppSelector((state) => state.food.filterValues);
+  const dropDownValue = useAppSelector((state) => state.food.dropDownValue);
+  const cards = useAppSelector((state) => state.food.cards);
+  return (
+    <>
+      {cards
+        ?.filter((el: ICard) => {
+          return filterValues.priceMin && filterValues.priceMax
+            ? el.price > filterValues.priceMin && el.price < filterValues.priceMax
+            : el;
+        })
+        ?.filter((el: ICard) => {
+          return filterValues.rating.length ? filterValues.rating.some((elem) => elem === el.popularity) : el;
+        })
+        ?.filter((el: ICard) => {
+          return filterValues.category.length
+            ? el.categoryPath.some((elem) =>
+                filterValues.category.some((elem2) => elem2?.toLowerCase() === elem.toLowerCase())
+              )
+            : el;
+        })
+        ?.filter((el: ICard) => {
+          return filterValues.brand.length ? filterValues.brand.some((elem) => elem === el.farm) : el;
+        })
+        ?.filter((el: ICard) => {
+          return dropDownValue.length === 1
+            ? el.categoryPath.some((el) => el === dropDownValue[0])
+            : dropDownValue.length === 2
+            ? dropDownValue[0] === el.farm && el.categoryPath.some((el) => el === dropDownValue[1].toLowerCase())
+            : el;
+        })
+        ?.filter((el: ICard) => {
+          return searchedValue.trim() ? el.name.toLowerCase().includes(searchedValue.trim().toLowerCase()) : el;
+        })
+        .map((el: ICard) => (
+          <Card {...el} key={el.id} />
+        ))}
+    </>
+  );
+};
+
```

Comment:
Are you sure that it's ok to have this in JSX??

Reply:
hm, I really thought, but the best thing I came up with is to move from cards to a separate component, maybe you have some ideas?

Reply:
I'm not sure that i want to put this calculations in redux , can create a function and pass arguments to it as for calculations in the utils folder, but also not good

Reply:
Describing functions in JSX it's bad practice. You can create separate functions, pass your array and call them step by step.
These functions could by in helpers folder or here, in this file - if you need. But not in JSX